examples/xlio_ultra_api_ping_pong.c

```diff
@@ -753,52 +643,35 @@ static void socket_event_cb(xlio_socket_t sock, uintptr_t userdata_sq, int event
 
 /*
  * Zero-copy completion callback
- * ============================
+ * =============================
  * Called when zero-copy send operations complete, allowing buffer reuse.
+ * Returns the buffer to the TX buffer pool.
  */
 static void socket_comp_cb(xlio_socket_t sock, uintptr_t userdata_sq, uintptr_t userdata_op)
 {
-    printf("Zero-copy send completed (socket userdata=%lx, op userdata=%lx)\n", userdata_sq,
-           userdata_op);
+    (void)sock;
+
+    char *buffer_ptr = (char *)userdata_op;
```

Comment:
as this is a customer-facing example, consider declaring all variables at the start of the function (traditional C style) consistently, OR use C99/C++ mixed declarations consistently. The current mix may confuse readers.